benchmark/src/main/java/com/datatorrent/benchmark/spillable/SpillableTestOperator.java

```diff
@@ -0,0 +1,179 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.benchmark.spillable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.apex.malhar.lib.state.spillable.SpillableArrayListImpl;
+import org.apache.apex.malhar.lib.state.spillable.SpillableByteArrayListMultimapImpl;
+import org.apache.apex.malhar.lib.state.spillable.SpillableByteMapImpl;
+import org.apache.apex.malhar.lib.state.spillable.SpillableStateStore;
+import org.apache.apex.malhar.lib.state.spillable.managed.ManagedStateSpillableStateStore;
+import org.apache.apex.malhar.lib.utils.serde.SerdeLongSlice;
+import org.apache.apex.malhar.lib.utils.serde.SerdeStringSlice;
+
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.Operator;
+import com.datatorrent.common.util.BaseOperator;
+
+public class SpillableTestOperator extends BaseOperator implements Operator.CheckpointNotificationListener
+{
+  public static transient final Logger logger = LoggerFactory.getLogger(SpillableTestOperator.class);
+  
+  public static final byte[] ID1 = new byte[] { (byte)1 };
+  public static final byte[] ID2 = new byte[] { (byte)2 };
+  public static final byte[] ID3 = new byte[] { (byte)3 };
+  
+  public SpillableByteArrayListMultimapImpl<String, String> multiMap;
+  
+  public ManagedStateSpillableStateStore store;
+  
+  public long totalCount = 0;
+  public transient long countInWindow;
+  public long minWinId = -1;
+  public long committedWinId = -1;
+  public long windowId;
+  
+  public SpillableByteMapImpl<Long, Long> windowToCount;
+  
+  public long shutdownCount = -1;
+  
+  public final transient DefaultInputPort<String> input = new DefaultInputPort<String>()
+      {
+        @Override
+        public void process(String tuple)
+        {
+          processTuple(tuple);
+        }
+      };
+      
+  public void processTuple(String tuple)
+  {
+    if(++totalCount == shutdownCount)
+      throw new RuntimeException("Test recovery. count = " + totalCount);
+    countInWindow++;
+    multiMap.put(""+windowId, tuple);
+  }
+  
+  @Override
+  public void setup(OperatorContext context)
+  {
+    super.setup(context);
+    if(windowToCount == null) {
```

Comment:
Please check maxAllowedViolations. Would be good to clean up all checkstyle violations in this module.